ui/src/components/Navigation.tsx

```diff
@@ -13,125 +12,76 @@ type Props = {
-                        name="applications"
-                      />{" "}
-                      Clients
-                    </NavLink>
-                  </li>
-                  <li className="p-side-navigation__item secondary">
-                    <NavLink
-                      className="p-side-navigation__link"
-                      to="/identity"
-                      title={`Identity list`}
-                    >
-                      <Icon
-                        className="is-light p-side-navigation__icon"
-                        name="user"
-                      />{" "}
-                      Identities
-                    </NavLink>
-                  </li>
-                  <li className="p-side-navigation__item secondary">
-                    <NavLink
-                      className="p-side-navigation__link"
-                      to="/schema"
-                      title={`Schema list`}
-                    >
-                      <Icon
-                        className="is-light p-side-navigation__icon"
-                        name="profile"
-                      />{" "}
-                      Schemas
-                    </NavLink>
-                  </li>
-                  <li className="p-side-navigation__item secondary">
-                    <GroupsLink
-                      className="p-side-navigation__link"
-                      baseURL="/"
-                      icon="user-group"
-                      iconIsLight
-                    />
-                  </li>
-                  <li className="p-side-navigation__item secondary">
-                    <RolesLink
-                      className="p-side-navigation__link"
-                      baseURL="/"
-                      icon="profile"
-                      iconIsLight
-                    />
-                  </li>
-                </ul>
-              </div>
-              <div className="p-side-navigation--icons is-dark p-side-navigation--user-menu">
-                <ul className="p-side-navigation__list">
-                  <li className="p-side-navigation__item">
-                    <span className="p-side-navigation__text">
-                      <Icon
-                        className="is-light p-side-navigation__icon"
-                        name="user"
-                      />{" "}
-                      {username}
-                    </span>
-                  </li>
-                  <li className="p-side-navigation__item">
-                    <Button
-                      appearance="link"
-                      className="p-side-navigation__link"
-                      onClick={() => {
-                        logout();
-                      }}
-                    >
-                      Logout
-                    </Button>
-                  </li>
-                </ul>
-              </div>
-            </div>
-          </div>
-        </div>
-      </nav>
+      <SideNavigation<NavLinkProps>
```

Comment:
I've never seen this before, is this legit?

Reply:
Hah yeah it's how you pass a TypeScript generic to a React component.

Reply:
oooh of course it's a type 🤦! I thought it was a JSX Component within a JSX Component definition... THIS makes more sense!